src/Apps/Order/Routes/Shipping2/Hooks/useHandleUserAddressUpdates.ts

```diff
@@ -0,0 +1,109 @@
+import { useShippingContext } from "Apps/Order/Routes/Shipping2/Hooks/useShippingContext"
+import { useCreateSavedAddress } from "Apps/Order/Routes/Shipping2/Mutations/useCreateSavedAddress"
+import { useDeleteSavedAddress } from "Apps/Order/Routes/Shipping2/Mutations/useDeleteSavedAddress"
+import { useUpdateSavedAddress } from "Apps/Order/Routes/Shipping2/Mutations/useUpdateSavedAddress"
+import {
+  FulfillmentType,
+  FulfillmentValues,
+  addressWithFallbackValues,
+  matchAddressFields,
+} from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
+import createLogger from "Utils/logger"
+
+const logger = createLogger(
+  "Order/Routes/Shipping/Hooks/useHandleUserAddressUpdates.tsx"
+)
+
+export const useHandleUserAddressUpdates = () => {
```

Comment:
Another bulk of behavior, which is some shared logic perfectly suited for a hook, so created it which we now import into the relevant locations.

Reply:
All these hooks should be easy to test now, as self-contained things independent of all else and easily mockable.

Reply:
Interesting... I wrote it this way very early on but then backed it out and kept only the mutations themselves in the hooks. no problems here, I guess for me it came down to a component owning its mutation